src/mocks/baseMocks/requestCreator.ts

```diff
@@ -54,6 +60,36 @@ export default function requestCreator<Resource extends BaseResource>(
           );
         }
 
+        if (relatedStores?.length) {
+          for (const relatedStore of relatedStores) {
+            const related: BaseResource[] = [];
+            const { relatedStoreName, relationReference } = relatedStore;
+
+            // We first get all data from the related store
+            const relatedDataList = await stores[
+              relatedStoreName
+            ].getListData();
+
+            // We filter the data to keep only what is related to
+            // this particular entity
+            relatedDataList.data.forEach((relatedData) => {
+              const relation =
+                relatedData.relationships?.[relationReference]?.data;
+              if (!Array.isArray(relation)) {
+                if (id === relation?.id) {
+                  related.push(relatedData);
+                }
+              }
+            });
+
+            // We update the response data with the relationships
+            data = {
+              ...data,
+              relationships: { [relatedStoreName]: { data: related } },
+            };
+          }
+        }
+
         const included = await getIncluded([data]);
```

Comment:
Here is how I propose we deal with one to many relationships like Project -> Roles.
Basically we add the optional relatedStores parameter here and if it is there we get the related data from the store we want (roles for example) and add it to the response.
@christopherjbaker let me know if it works for you or if you have other suggestions. If it works I'll clean up the code and test it with other fixtures. I tested with project roles and it works.

Reply:
Actually there might be overlap with `getIncluded`...